src/lib/optimizer/strategy/logical_reduction_rule.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <unordered_map>
+
+#include "abstract_rule.hpp"
+#include "expression/abstract_expression.hpp"
+#include "expression/logical_expression.hpp"
+
+namespace opossum {
+
+class AbstractLQPNode;
+
+/**
+ * This rule tries to simplify logical expressions, so that they can be processed faster.
+ *
+ * Reverse Distributivity
+ *   Expressions of the form `(a AND b) OR (a AND c)` are rewritten as `a AND (b OR c)`.
```

Comment:
"extract common expressions" is so general that reading it gives me, personally, no clue as to what is happening. "Distributivity" on the other hand is something everyone has heard of, so it gives you at least some sort of idea what operation is being applied to the expression.

(I don't think "reverse/reduce distributivity" is a great term either, especially since it's just the term for a property. Maybe "reduce_with_law_of_distributivity" or something like that?)